### PR TITLE
JWT Auth

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

The backend has been adapted (in the auth-tokens branch) to remove all references to `express-session` and `csurf`.

JWT's are now issued and verified using the new authentication middleware. All requests must now be made including the header:

```
Authorization: Bearer <token>
```

Tokens can be issued from the login endpoints directly or via the Facebook Login plugin which places the token in localStorage for you.

### CODE REVIEW

File: services/passport.js [hunk 1]
```diff
@@ -38,38 +40,30 @@ const HandleAuthCallback = (req, res, next) => (err, user) => {
...
  */
 const HandleAuthPopupCallback = (req, res, next) => (err, user) => {
   if (err) {
-    return res.render('auth-callback', {err: JSON.stringify(err), data: null});
+    return res.render('auth-callback', {auth: JSON.stringify({err, data: null})});
   }
 
   if (!user) {
-    return res.render('auth-callback', {err: JSON.stringify(errors.ErrNotAuthorized), data: null});
+    return res.render('auth-callback', {auth: JSON.stringify({err, data: null})});
```
Comment (alinhle) [inline services/passport.js]: I think you need to set `err` to something meaningful here.

File: plugins/coral-plugin-facebook-auth/server/router.js [hunk 2]
```diff
@@ -15,6 +15,6 @@ module.exports = (router) => {
   router.get('/api/v1/auth/facebook/callback', (req, res, next) => {
 
     // Perform the facebook login flow and pass the data back through the opener.
-    passport.authenticate('facebook', HandleAuthPopupCallback(req, res, next))(req, res, next);
+    passport.authenticate('facebook', {session: false}, HandleAuthPopupCallback(req, res, next))(req, res, next);
```
Comment (alinhle) [inline plugins/coral-plugin-facebook-auth/server/router.js:18]: what does `session: false` mean?

**Comment by alinhle** [comment]
This looks really nice so far!

File: plugins/coral-plugin-facebook-auth/server/router.js (hunk 2 above)
Comment (wyattjoh) [reply plugins/coral-plugin-facebook-auth/server/router.js:18]: This is disabling the underlying session features on Passport.

**Comment by okbel** [comment]
Frontend ready!

**Review by okbel — APPROVED** [review]

**Pull request merged**